### ISSUE TITLE
Windows commandline input doesn't work

### ISSUE THREAD

**Issue by irvnriir** [GitCredentialManager/git-credential-manager#634]
**Which version of GCM are you using?**

From a terminal, run `git credential-manager-core --version` and paste the output.

2.0.498+7ad55fb809

**Which Git host provider are you trying to connect to?**

* [ ] Azure DevOps
* [ ] Azure DevOps Server (TFS/on-prem)
* [x] GitHub

---

---

.cmd
```
@echo off
setlocal enableExtensions enableDelayedExpansion
 
(
	echo(protocol=https
	echo(host=github.com
) | git credential-manager-core erase
 
(
	echo(protocol=https
	echo(host=github.com
	echo(username=<<_>>
	echo(password=<<Personal Access Token>>
) | git credential-manager-core store
 
pause
```

**Expected behavior**

**Actual behavior**

`fatal: value` each `git credential-manager-core` .

when some of the parameter lines are missed, it shows error with the 1st missed var name, so it doesn't seem an input/cmd.exe issue .

ik `cmdkey` can be used, the issue still present .

**Logs**

checked, doesn't have anything useful .

**Comment by dscho** [comment]
This really is not a Git Credential Manager bug, but rooted in the challenge to provide proper input in a `.cmd` script. In particular, the output of your `( echo ... )` contains not only Carriage Returns at the end of line, but also a trailing space!

I have no idea where that space is coming from, but it sure ain't coming from Git Credential Manager.

You can work around this by abusing Git's `alias` functionality to call the `tr.exe` utility that is currently included in Git for Windows:

```
(
	echo(protocol=https
	echo(host=fake.com
	echo(username=abc
	echo(password=def
) | git -c alias.t="!tr -d ' '" t | git credential-manager-core store
```

**Comment by irvnriir** [comment]
for me it says "tr" not found, though it does present in the direc

**Comment by dscho** [comment]
> for me it says "tr" not found

That is funny because this tool is specifically included in Git for Windows. Maybe you have a "special" setup where you run with MinGit? But no, even MinGit includes a `tr.exe`.

> though it does present in the git direc

Hmm. I found this report already pretty short on details upon reading the initial report (i.e. already on the other side of succinct), but this is no longer only very curt, it is very incomplete.

**Comment by irvnriir** [comment]
idk how this could be reasonably more detailed
![Capture](https://user-images.githubusercontent.com/13949704/159346869-01843261-109c-47a0-87b7-aa0c42243af0.PNG)


the same is with the thread's issue

p.s. unless you are joking about "direc"

**Comment by dscho** [comment]
If it says `'tr' is not a git command` (which is a lot more detail than you offered originally, by the way), it suggests that you missed the `!` in the alias. But I cannot say for sure because the command-line was not included in the response.

**Comment by irvnriir** [comment]
sorry for the confusion . but i didn't write `` ` `` + `not found` + `` ` `` , the error means command is "not found" . i assume you considered its about path, but the issue happened because your command essentially needs to target a command, not path . the amount of info i work with is exhaustive, its very hard to track such little formatting details .

good tip about `!` , with `setlocal enableExtensions enableDelayedExpansion` its `git -c alias.t="^!tr -d ' '" t` .

**Comment by irvnriir** [comment]
Edit: you actually followed my consideration that its about path XD . sorry x2 . but still i didn't know what it targets, but the error's meaning is the same .

**Comment by irvnriir** [comment]
Edit: you actually followed my consideration that its about path XD . sorry x2 . but still i didn't know what it targets, but the error's meaning is the same .

**Comment by irvnriir** [comment]
(accident)

**Comment by ldennington** [comment]
Here is how I worked around the issue on my machine:

Open cmd prompt.
`setx path "%PATH%;C:\Program Files\Git\usr\bin"`

Open new cmd prompt.
Run modified script (just uses `tr` rather than aliasing)
```cmd
(
	echo(protocol=https
	echo(host=fake.com
) | tr -d ' ' | git credential-manager-core erase
 
(
	echo(protocol=https
	echo(host=fake.com
	echo(username=abc
	echo(password=def
) | tr -d ' ' | git credential-manager-core store
 
pause
````

Please let us know if you continue to experience issues.

**Comment by irvnriir** [comment]
@ldennington already solved with
> good tip about `!` , with `setlocal enableExtensions enableDelayedExpansion` its `git -c alias.t="^!tr -d ' '" t` .

this is a workaround, not a solution to the original issue .

**Comment by irvnriir** [comment]
@dscho output of `( echo ... )` doesn't contain trailing spaces . can the issue be related to git handling of CRLF line endings ?

**Comment by ldennington** [comment]
My bad for missing that. However, I'm going to leave this issue closed, as I believe we have determined it is not GCM related and we've been able to work around it.

**Comment by dscho** [comment]
No, the issue is that the `echo` calls add those trailing spaces for some reason, not Git doing anything wrong. The `tr` call is just a workaround for that.

And yes, I agree, this is neither a Git nor a GCM problem.
